src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1520,8 +1520,8 @@ public override T GetFieldValue<T>(int ordinal)
             {
                 if (Connector.State != ConnectorState.Broken)
                 {
-                    var writtenBytes = Buffer.ReadPosition - position;
-                    var remainingBytes = ColumnLen - writtenBytes;
+                    var readBytes = Buffer.ReadPosition - position;
+                    var remainingBytes = ColumnLen - readBytes;
                     if (remainingBytes > 0)
                         Buffer.Skip(remainingBytes, false).GetAwaiter().GetResult();
                 }
```

Comment:
Do we really need to catch and handle exception thrown inside type handlers for the non-sequential case, and make sure skip to end? Can't we leave it at any ReadPosition? When the next column or row is read, a new seek to column (or seek to row end if we move to the next row) is done anyway.